src/main/java/io/anserini/rerank/lib/NewsBackgroundLinkingReranker.java

```diff
@@ -0,0 +1,144 @@
+
+import static io.anserini.index.generator.LuceneDocumentGenerator.FIELD_BODY;
+import static io.anserini.index.generator.LuceneDocumentGenerator.FIELD_ID;
+
+
+/*
+* TREC News track Background Linking task postprocessing.
+* Near-duplicate documents (similar/same with the query docid) will be removed by comparing
+* their cosine similarity with the query docid.
+ */
+public class NewsBackgroundLinkingReranker implements Reranker {
+  @Override
+  public ScoredDocuments rerank(ScoredDocuments docs, RerankerContext context) {
+    IndexReader reader = context.getIndexSearcher().getIndexReader();
+    String queryDocId = context.getQueryDocId();
+    final Map<String, Long> queryTermsMap = convertDocVectorToMap(reader, queryDocId);
+    
+    List<Map<String, Long>> docsVectorsMap = new ArrayList<>();
+    for (int i = 0; i < docs.documents.length; i++) {
+      String docid = docs.documents[i].getField(FIELD_ID).stringValue();
+      docsVectorsMap.add(convertDocVectorToMap(reader, docid));
+    }
+    
+    // remove the duplicates: 1. the same doc with the query doc 2. duplicated docs in the results
+    Set<Integer> forbiddenDocIdsIndex = new HashSet<>();
+    for (int i = 0; i < docs.documents.length; i++) {
+      if (forbiddenDocIdsIndex.contains(i)) continue;
+      String docid = docs.documents[i].getField(FIELD_ID).stringValue();
+      if (computeCosineSimilarity(queryTermsMap, docsVectorsMap.get(i)) >= 0.9) {
+        forbiddenDocIdsIndex.add(i);
+        continue;
+      }
+      for (int j = i+1; j < docs.documents.length; j++) {
+        if (computeCosineSimilarity(docsVectorsMap.get(i), docsVectorsMap.get(j)) >= 0.9) {
+          forbiddenDocIdsIndex.add(j);
+        }
+      }
+    }
+  
+    ScoredDocuments scoredDocs = new ScoredDocuments();
+    int resSize = docs.documents.length - forbiddenDocIdsIndex.size();
+    scoredDocs.documents = new Document[resSize];
+    scoredDocs.ids = new int[resSize];
+    scoredDocs.scores = new float[resSize];
+    int idx = 0;
+    for (int i = 0; i < docs.documents.length; i++) {
+      if (!forbiddenDocIdsIndex.contains(i)) {
+        scoredDocs.documents[idx] = docs.documents[i];
+        scoredDocs.scores[idx] = docs.scores[i];
+        scoredDocs.ids[idx] = docs.ids[i];
+        idx++;
+      }
+    }
+  
+    return scoredDocs;
+  }
+  
+  private Map<String, Long> convertDocVectorToMap(IndexReader reader, String docid) {
+    Map<String, Long> m = new HashMap<>();
+    try {
+      Terms terms = reader.getTermVector(NewsBackgroundLinkingTopicReader.convertDocidToLuceneDocid(reader, docid), FIELD_BODY);
+      TermsEnum it = terms.iterator();
+      while (it.next() != null) {
+        String term = it.term().utf8ToString();
+        long tf = it.totalTermFreq();
+        m.put(term, tf);
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    return m;
+  }
+  
+  private double dotProduct(Map<String, Long> profile1, Map<String, Long> profile2) {
```

Comment:
we have similar methods in `io.anserini.util.FeatureVector` - does it make sense to combine functionality?